packages/jsActions/nanoflow-actions-native/src/geolocation/GetCurrentLocation.ts

```diff
@@ -32,15 +35,28 @@ export async function GetCurrentLocation(
 ): Promise<mendix.lib.MxObject> {
     // BEGIN USER CODE
 
-    if (navigator && navigator.product === "ReactNative" && !navigator.geolocation) {
-        (navigator.geolocation as any) = Geolocation;
+    let rnGeolocation: GeolocationServiceStatic | GeolocationStatic;
+
+    if (navigator && navigator.product === "ReactNative") {
+        if (NativeModules.RNFusedLocation) {
+            const geolocationService = await import("react-native-geolocation-service");
+            rnGeolocation = geolocationService.default;
+        } else if (NativeModules.RNCGeolocation) {
+            rnGeolocation = Geolocation;
+        } else {
+            return Promise.reject(new Error("Geolocation module couldn't find"));
+        }
+    } else if (navigator && navigator.geolocation) {
+        rnGeolocation = navigator.geolocation;
+    } else {
+        return Promise.reject(new Error("Navigator couldn't find"));
```

Comment:
This error string could be improved. What about, "Geolocation module could not be found"